extensions/amp-lightbox-viewer/0.1/amp-lightbox-viewer.js

```diff
@@ -202,6 +202,24 @@ export class AmpLightboxViewer extends AMP.BaseElement {
     this.container_.appendChild(mask);
   }
 
+  /**
+   * Return a cleaned clone of the given element for building
+   * carousel slides with.
+   * @param {!Element} element
+   * @private
+   */
+  cloneLightboxableElement_(element) {
+    const deepClone = !element.classList.contains(
+        'i-amphtml-element');
+    let clonedNode = element.cloneNode(deepClone);
+    if (element.tagName == 'FIGURE') {
+      clonedNode = element.getElementsByTagName('amp-img')[0]
```

Comment:
What will happen if a lightboxable `<figure>` doesn't have an `<amp-img>`? Or is a lightboxable figure guaranteed to have an `<amp-img>`? I couldn't find that in the code but I didn't spend very long looking

Reply:
Yup, this is all pretty sketch. I'll add some validation code. We had some debate on whether or not to allow the `<img>` tag, or to force `<amp-img>`. In this case I'll allow either and throw a user assertion error if neither is present.